libvast/src/system/importer.cpp

```diff
@@ -38,44 +39,73 @@ importer_state::importer_state(caf::event_based_actor* self_ptr)
 }
 
 importer_state::~importer_state() {
-  write_state();
+  write_state(write_mode::with_next);
 }
 
 caf::error importer_state::read_state() {
-  auto file = dir / "next_id";
+  auto file = dir / "current_id_block";
   if (exists(file)) {
     VAST_VERBOSE(self, "reads persistent state from", file);
-    std::ifstream available{to_string(file)};
-    available >> next_id;
-    if (!available.eof())
-      VAST_ERROR(self, "got an invalidly formatted persistence file:", file);
+    std::ifstream state_file{to_string(file)};
+    state_file >> current.end;
+    state_file >> current.next;
+    // TODO:
+    if (!state_file) {
+      VAST_WARNING(self, "did not find next id position from the state file; "
+                         "detected an irregular shutdown");
+      current.next = current.end;
+    }
   } else {
     VAST_VERBOSE(self, "did not find a state file at", file);
+    current.end = 0;
+    current.next = 0;
   }
-  return caf::none;
+  return get_next_block();
 }
 
-caf::error importer_state::write_state() {
+caf::error importer_state::write_state(write_mode mode) {
   if (!exists(dir)) {
     auto result = mkdir(dir);
     if (!result)
       return std::move(result.error());
   }
-  std::ofstream available{to_string(dir / "next_id")};
-  available << next_id;
-  VAST_VERBOSE(self, "persisted id space caret at", next_id);
+  std::ofstream state_file{to_string(dir / "current_id_block")};
+  state_file << current.end;
+  if (mode == write_mode::with_next) {
+    state_file << " " << current.next;
+    VAST_VERBOSE(self, "persisted id block [", current.next, ",", current.end,
+                 ")");
+  } else {
+    VAST_VERBOSE(self, "persisted id block boundary at", current.end);
+  }
   return caf::none;
 }
 
+caf::error importer_state::get_next_block() {
+  using namespace si_literals;
+  while (current.next >= current.end)
+    current.end += 8_Mi;
```

Comment:
How about putting this as a `constexpr` parameters in namespace `defaults`?

Reply:
That would suggest that it can be changed, which it cannot. It's only used in a single place, so I don't really see a need for obfuscating this value.

Reply:
I agree with @dominiklohmann, we should not make this configurable.